sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/aio/_data_lake_lease_async.py

```diff
@@ -30,8 +30,7 @@ class DataLakeLeaseClient:  # pylint: disable=client-accepts-api-version-keyword
         The client of the file system, directory, or file to lease.
     :type client:
         ~azure.storage.filedatalake.aio.FileSystemClient or
-        ~azure.storage.filedatalake.aio.DataLakeDirectoryClient or
-        ~azure.storage.filedatalake.aio.DataLakeFileClient
+        ~azure.storage.filedatalake.aio.PathClient
```

Comment:
Is this correct (omitting ` ~azure.storage.filedatalake.aio.DataLakeDirectoryClient` and `~azure.storage.filedatalake.aio.DataLakeFileClient`)?
Both clients inherit from `PathClient`, so I would expect (and since most users don't actually keep around a specific `PathClient` base client instance) that you should be able to just pass these "parent" clients. 

Maybe just do a quick ad-hoc test for sanity? 

Additionally, if this is truly the case that it specifically requires the base client variant, you need to correct the error raised below.